src/dag/get.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+const promisify = require('promisify-es6')
+const CID = require('cids')
+const waterfall = require('async/waterfall')
+const block = require('../block')
+
+module.exports = (send) => {
+  return promisify((cid, path, options, callback) => {
+    if (typeof path === 'function') {
+      callback = path
+      path = undefined
+    }
+
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+
+    options = options || {}
+
+    if (CID.isCID(cid)) {
+      cid = cid.toBaseEncodedString()
+    }
+
+    if (typeof cid === 'string') {
+      const split = cid.split('/')
+      cid = split[0]
+      split.shift()
+
+      if (split.length > 0) {
+        path = split.join('/')
+      } else {
+        path = '/'
+      }
+    }
+
+    waterfall([
+      cb => {
+        send({
+          path: 'dag/resolve',
+          args: cid + '/' + path,
+          qs: options
+        }, cb)
+      },
+      (resolved, cb) => {
+        block(send).get(new CID(resolved['Cid']['/']), (err, blk) => cb(err, blk, resolved['RemPath']))
+      },
+      (blk, path, cb) => {
+        if (blk.cid.codec === 'dag-cbor') {
+          dagCBOR.resolver.resolve(blk, path, cb)
+        }
+        if (blk.cid.codec === 'dag-pb') {
+          dagPB.resolver.resolve(blk, path, cb)
+        }
+      }
+    ], callback)
```

Comment:
go-ipfs side is done. Now that i think of it, is there any reason the code for put which was here earlier didn't just pass the data as json to go-ipfs? Passing json directly instead of serializing it here would allow to support every encoding go side supports without additional work here.

The same is the case for `get` with the difference that using go-ipfs dag get doesn't return `remainderPath`(I don't think there is any case where it wouldn't be empty). Does the value passed to callback  in `get` implement any API or is it much like just like parsing json from `ipfs dag get`?